util/lua/lua.go

```diff
@@ -39,6 +39,8 @@ type ResourceHealthOverrides map[string]appv1.ResourceOverride
 func (overrides ResourceHealthOverrides) GetResourceHealth(obj *unstructured.Unstructured) (*health.HealthStatus, error) {
 	luaVM := VM{
 		ResourceOverrides: overrides,
+		// check if the usage of lua standard library is enabled
+		UseOpenLibs: overrides.isOpenLibsEnabled(obj),
 	}
 	script, err := luaVM.GetHealthScript(obj)
```

Comment:
The `UseOpenLibs` should be `true` for all built-in scripts. 

I think you might return `isBuiltIn` boolean from `GetHealthScript` and use it to enable/disable `UseOpenLibs`

https://github.com/argoproj/argo-cd/blob/45713b1ffca5944143a68435dff43eda071286ae/util/lua/lua.go#L130

Reply:
Thanks! Made the changes